### PR TITLE
Check numRuns in HashStringAllocator::checkConsistency

### PR DESCRIPTION
**Pull request by waitinfuture**
Since AllocationPool directly passes the needed pages as minimal size class
(refer the commit message of
https://github.com/facebookincubator/velox/pull/1906), the
```allocation->numRuns()``` should always be 1 in HashStringAllocator's pool_.
Add the check in ```HashStringAllocator::checkConsistency()```.

If ```allocation->numRuns()``` is more than 1, ```AllocationPool::newRunImpl```
can just return the next run, which might not satisfy required ```numPages```.

### CODE REVIEW

File: velox/common/memory/HashStringAllocator.cpp [hunk 1]
```diff
@@ -402,6 +402,7 @@ void HashStringAllocator::checkConsistency() const {
   VELOX_CHECK_EQ(pool_.numLargeAllocations(), 0);
   for (auto i = 0; i < pool_.numSmallAllocations(); ++i) {
     auto allocation = pool_.allocationAt(i);
+    VELOX_CHECK_EQ(allocation->numRuns(), 1);
```
Comment (mbasmanova) [inline velox/common/memory/HashStringAllocator.cpp:405]: If this check is valid, then let's also simplify the loop that follows. That loop runs just once for runIndex of 0.

**Review by mbasmanova — COMMENTED** [review]
@waitinfuture Would you shorten the PR title so that it fits?

File: velox/common/memory/HashStringAllocator.cpp (hunk 1 above)
Comment (waitinfuture) [reply velox/common/memory/HashStringAllocator.cpp:405]: > 

Updated, thanks.

**Review by mbasmanova — APPROVED** [review]
@waitinfuture Thanks.

File: velox/common/memory/HashStringAllocator.cpp [hunk 2]
```diff
@@ -402,37 +402,35 @@ void HashStringAllocator::checkConsistency() const {
...
-          auto continued = getNextContinued(header);
-          VELOX_CHECK(!continued->isFree());
+    VELOX_CHECK_EQ(allocation->numRuns(), 1);
+    auto run = allocation->runAt(0);
+    auto size = run.numBytes() - sizeof(Header);
+    bool previousFree = false;
+    auto end = reinterpret_cast<Header*>(run.data<char>() + size);
+    auto header = run.data<Header>();
+    while (header != end) {
+      VELOX_CHECK(header >= run.data<Header>());
```
Comment (xiaoxmeng) [inline velox/common/memory/HashStringAllocator.cpp:412]: nit: could you please change to use VELOX_CHECK_GE/LT/GT for comparison checks here?

Comment (waitinfuture) [reply velox/common/memory/HashStringAllocator.cpp]: > nit: could you please change to use VELOX_CHECK_GE/LT/GT for comparison checks here?

Updated, thanks @xiaoxmeng

**Review by mbasmanova — APPROVED** [review]
@waitinfuture Thank you for additional enhancements.

**Pull request closed without merging**